File: Adarsh/RotateArray.java
class Solution {
    public void rotate(int[] nums, int k) {
        List<Integer> l = new ArrayList<Integer>();
        
        int len = nums.length;
        for(int i = 0; i < len; i++) {
            l.add(nums[i]);
        }
    
        int s = 0;
        while(k > 0) {
            k--;
            s = l.get(l.size()-1);
            l.remove(l.size()-1);
            l.add(0, s);
        }
        
        for(int i = 0; i < len ; i++)
            nums[i] = l.get(i);
    }
}

File: Pranai/Arrays/Rotate_Array.c
void rotate(int* nums, int numsSize, int k){
    if(k>numsSize)
        k-=numsSize;
    if(k>0){
    int a[k],i,j=0;
    for(i=0;i<k;i++)
        a[i]=0;
    for(i=numsSize-k;i<numsSize;i++){
        a[j]=nums[i];
        j++;
    }
    for(i=numsSize-k-1;i>=0;i--)
        nums[i+k]=nums[i];
    
    for(i=0;i<k;i++)
        nums[i]=a[i];
    }
}
